tests/production/CloudStatusProductionTests.js

```diff
@@ -0,0 +1,89 @@
+/**
+GPII Production Config tests
+
+Requirements:
+* an internet connection
+* a cloud based flow manager
+* a preferences server
+* a CouchDB server
+
+---
+
+Copyright 2015 Raising the Floor - International
+Copyright 2018, 2019 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.require("%gpii-universal");
+
+gpii.loadTestingSupport();
+
+fluid.registerNamespace("gpii.tests.productionConfigTesting");
+
+require("../shared/DevelopmentTestDefs.js");
+require("./ProductionTestsUtils.js");
+
+// Flowmanager tests for:
+// /user/%gpiiKey/login and /user/%gpiiKey/logout (as defined in gpii.tests.development.testDefs),
+// /health,
+// /ready,
+gpii.tests.productionConfigTesting.testDefs = fluid.transform(gpii.tests.development.testDefs, function (testDefIn) {
+    var testDef = fluid.extend(true, {}, testDefIn, {
```

Comment:
All of this material should be expressed as an addon grade rather than stuck here in code. Given the argument to gpii.test.push just consists of static material there's no value (actually negative value) to writing it in code.
Actually the whole existence of gpii.test.push seems like a strange historical artefact, and give it is used in just one place we had better just get rid of it.

Reply:
There's another use of `gpii.test.push()` in [UntrustedPSPIntegrationTests.js](https://github.com/GPII/universal/blob/master/tests/UntrustedPSPIntegrationTests.js#L328).  Its inverse, `gpii.test.unshift(),` is used [there](https://github.com/GPII/universal/blob/master/tests/UntrustedPSPIntegrationTests.js#L324) too.

I think that's all of the uses.  I don't see either function used in the windows tests.  I haven't looked at the linux nor android repositories.  Anywhere else?

Reply:
@amb26 Replaced use of `gpii.test.push()` with grades, with these [two](https://github.com/GPII/universal/pull/718#commits-pushed-18a062c) [commits](https://github.com/GPII/universal/pull/718#commits-pushed-b490985).